src/modules/ekf2/ekf2_main.cpp

```diff
@@ -920,6 +980,131 @@ void Ekf2::task_main()
 				}
 			}
 
+			// publish estimator status
+			{
+				struct estimator_status_s status = {};
+				status.timestamp = now;
+				_ekf.get_state_delayed(status.states);
+				_ekf.get_covariances(status.covariances);
+				_ekf.get_gps_check_status(&status.gps_check_fail_flags);
+				_ekf.get_control_mode(&status.control_mode_flags);
+				_ekf.get_filter_fault_status(&status.filter_fault_flags);
+				_ekf.get_innovation_test_status(&status.innovation_check_flags, &status.mag_test_ratio,
+								&status.vel_test_ratio, &status.pos_test_ratio,
+								&status.hgt_test_ratio, &status.tas_test_ratio,
+								&status.hagl_test_ratio);
+
+				status.pos_horiz_accuracy = lpos.eph;
+				status.pos_vert_accuracy = lpos.epv;
+				_ekf.get_ekf_soln_status(&status.solution_status_flags);
+				_ekf.get_imu_vibe_metrics(status.vibe);
+
+				if (_estimator_status_pub == nullptr) {
+					_estimator_status_pub = orb_advertise(ORB_ID(estimator_status), &status);
+
+				} else {
+					orb_publish(ORB_ID(estimator_status), _estimator_status_pub, &status);
+				}
+
+				/*
+				 * Check if conditions are OK to save learned magnetometer bias values after 3min of the following:
+				 * Armed, In air, using 3-axis mag fusion, no filter faults
+				 * Also check for changes in Mag ID, but do not apply 3-min rule to this check to allow for
+				 * occasional in-flight mag sensor timeouts which can cause switching from primary to secondary mag
+				*/
+				bool mag_cal_active = status.control_mode_flags & (1 << 5);
+
+				if (vehicle_land_detected.landed
+				    || (vehicle_status.arming_state != vehicle_status_s::ARMING_STATE_ARMED)
+				    || (status.filter_fault_flags != 0)
+				    || !mag_cal_active) {
+					_last_invalid_magcal_us = now;
+
+				} else if (((now - _last_invalid_magcal_us) > 180E6)
```

Comment:
This should be `180*1000*1000ULL` or `180*1000000ULL`, because 180E6 is a `double` (I know it looks less readable)